lib/buffer.js

```diff
@@ -246,18 +247,30 @@ function fromArrayLike(obj) {
   const length = obj.length;
   const b = allocate(length);
   for (var i = 0; i < length; i++)
-    b[i] = obj[i] & 255;
+    b[i] = obj[i];
   return b;
 }
 
 function fromArrayBuffer(obj, byteOffset, length) {
+  if (!isArrayBuffer(obj))
+    throw new TypeError('argument is not an ArrayBuffer');
+
   byteOffset >>>= 0;
 
-  if (typeof length === 'undefined')
-    return binding.createFromArrayBuffer(obj, byteOffset);
+  const maxLength = obj.byteLength - byteOffset;
+
+  if (maxLength <= 0)
```

Comment:
I think this should be `<`. I know this is taken directly from the C++ code, but I think 85ab4a5f1281c actually introduced a regression here:

Node v5.9.1:
```js
> new Buffer(new Uint8Array().buffer)
<Buffer >
```

Node v5.10.1:
```js
> new Buffer(new Uint8Array().buffer)
RangeError: 'offset' is out of bounds
```

… @jasnell?